In [1]:
import pandas as pd
# Librerías usadas para estandarizar la información
import repositories.dicts as dicts
import repositories.utils as utils
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


# Declaring the names of the columns we're interested in

In [2]:
columnas=['DP','DPNOM','DPMP','MPIO','AÑO','ÁREA GEOGRÁFICA']+[str(5*i)+"_"+str(5*(i+1)) for i in range(20)]+["Mujeres_100"]


# Reading the population by genre an municipalities for the period considered

df_1995_2004=pd.DataFrame(pd.read_excel("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.xlsx"))
df_2004_2017=pd.DataFrame(pd.read_excel("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.xlsx"))
df_2018=pd.DataFrame(pd.read_excel("anexo-proyecciones-poblacion-Municipal_2018-2026.xlsx"))

In [3]:
df_1995_2004=pd.DataFrame(pd.read_excel("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.xlsx"))

In [3]:
#df_1995_2004=pd.DataFrame(pd.read_csv("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.csv"))
df_1995_2004=pd.DataFrame(pd.read_excel("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.xlsx"))
df_2004_2017=pd.DataFrame(pd.read_csv("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.csv"))
df_2018=pd.DataFrame(pd.read_csv("data/anexo-proyecciones-poblacion-Municipal_2018-2026.csv"))

df_1995_2004.to_csv("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.csv")
df_2004_2017.to_csv("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.csv")
df_2018.to_csv("anexo-proyecciones-poblacion-Municipal_2018-2026.csv")



# Standardizing the names and codes of the municipalities, filtering data associated to the whole municipalities.

In [4]:
df_1995_2004=df_1995_2004[df_1995_2004['ÁREA GEOGRÁFICA']=="Total"].copy()
df_1995_2004.DPMP=df_1995_2004.DPMP.apply(lambda x: str(x).zfill(5))

In [5]:
df_2004_2017=df_2004_2017[df_2004_2017['ÁREA GEOGRÁFICA']=="Total"].copy()
df_2004_2017.DPMP=df_2004_2017.DPMP.apply(lambda x: str(x).zfill(5))

In [6]:
df_2018=df_2018[df_2018['ÁREA GEOGRÁFICA']=="Total"].copy()
df_2018.DPMP=df_2018.DPMP.apply(lambda x: str(x).zfill(5))


# Selecting data associated to woman populations during the time horizon considered

In [7]:
for i in range(20): 
    if i<19:
        print(["Mujeres_"+str(j) for j in range(i*5,(i+1)*5)])
        df_1995_2004[str(5*i)+"_"+str(5*(i+1))]=sum(df_1995_2004["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))
    else:
        print(["Mujeres_"+str(j) for j in range(i*5,(i+1)*5)])
        df_1995_2004[str(5*i)+"_"+str(5*(i+1))]=sum(df_1995_2004["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))#+df_1995_2004["Mujeres_100"]        

['Mujeres_0', 'Mujeres_1', 'Mujeres_2', 'Mujeres_3', 'Mujeres_4']
['Mujeres_5', 'Mujeres_6', 'Mujeres_7', 'Mujeres_8', 'Mujeres_9']
['Mujeres_10', 'Mujeres_11', 'Mujeres_12', 'Mujeres_13', 'Mujeres_14']
['Mujeres_15', 'Mujeres_16', 'Mujeres_17', 'Mujeres_18', 'Mujeres_19']
['Mujeres_20', 'Mujeres_21', 'Mujeres_22', 'Mujeres_23', 'Mujeres_24']
['Mujeres_25', 'Mujeres_26', 'Mujeres_27', 'Mujeres_28', 'Mujeres_29']
['Mujeres_30', 'Mujeres_31', 'Mujeres_32', 'Mujeres_33', 'Mujeres_34']
['Mujeres_35', 'Mujeres_36', 'Mujeres_37', 'Mujeres_38', 'Mujeres_39']
['Mujeres_40', 'Mujeres_41', 'Mujeres_42', 'Mujeres_43', 'Mujeres_44']
['Mujeres_45', 'Mujeres_46', 'Mujeres_47', 'Mujeres_48', 'Mujeres_49']
['Mujeres_50', 'Mujeres_51', 'Mujeres_52', 'Mujeres_53', 'Mujeres_54']
['Mujeres_55', 'Mujeres_56', 'Mujeres_57', 'Mujeres_58', 'Mujeres_59']
['Mujeres_60', 'Mujeres_61', 'Mujeres_62', 'Mujeres_63', 'Mujeres_64']
['Mujeres_65', 'Mujeres_66', 'Mujeres_67', 'Mujeres_68', 'Mujeres_69']
['Mujeres_70', '

In [8]:
for i in range(20): 
    if i<19:
        df_2004_2017[str(5*i)+"_"+str(5*(i+1))]=sum(df_2004_2017["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))
    else:
        df_2004_2017[str(5*i)+"_"+str(5*(i+1))]=sum(df_2004_2017["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))#+df_2004_2017["Mujeres_100"]

In [9]:
for i in range(20): 
    if i<20:
        df_2018[str(5*i)+"_"+str(5*(i+1))]=sum(df_2018["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))
        
    else:
        df_2018[str(5*i)+"_"+str(5*(i+1))]=sum(df_2018["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))#+df_2018["Mujeres_100"]

# Selecting the subset of columns we are interested in

In [21]:
df_1995_2004=df_1995_2004[columnas]
df_2004_2017=df_2004_2017[columnas]
df_2018=df_2018[columnas]

# Concatenating dataframes and saving it to an external file for further processing (if necessary)

In [22]:
df=pd.concat([df_1995_2004,df_2004_2017,df_2018])
df

DP      DPNOM   DPMP      MPIO   AÑO ÁREA GEOGRÁFICA    0_5   5_10  \
0       5  Antioquia  05001  Medellín  1995           Total  87865  85949   
1       5  Antioquia  05001  Medellín  1996           Total  88320  87597   
2       5  Antioquia  05001  Medellín  1997           Total  88319  88993   
3       5  Antioquia  05001  Medellín  1998           Total  88162  90335   
4       5  Antioquia  05001  Medellín  1999           Total  87698  91439   
...    ..        ...    ...       ...   ...             ...    ...    ...   
10093  99    Vichada  99773  Cumaribo  2022           Total   5136   5077   
10094  99    Vichada  99773  Cumaribo  2023           Total   5144   4985   
10095  99    Vichada  99773  Cumaribo  2024           Total   5021   4885   
10096  99    Vichada  99773  Cumaribo  2025           Total   5046   4810   
10097  99    Vichada  99773  Cumaribo  2026           Total   5081   4763   

       10_15  15_20  ...  55_60  60_65  65_70  70_75  75_80  80_85  85_90  \
0      84068  80847  ...  29407  24672  20169  14980  10543   6477   3342   
1      84741  83325  ...  30807  25075  20797  15622  11147   6839   3495   
2      85300  85540  ...  32335  25539  21392  16316  11728   7158   3705   
3      86309  87523  ...  34023  26166  21949  17021  12315   7430   3924   
4      87795  89112  ...  35824  26943  22456  17727  12577   7974   4099   
...      ...    ...  ...    ...    ...    ...    ...    ...    ...    ...   
10093   5009   4613  ...    946    716    494    345    193    112     50   
10094   4974   4667  ...    970    748    517    370    207    121     56   
10095   5024   4783  ...    982    768    539    378    210    104     41   
10096   4994   4835  ...   1010    792    572    393    231    108     43   
10097   4961   4872  ...   1049    811    605    409    252    113     47   

       90_95  95_100  Mujeres_100  
0       1573     582          250  
1       1602     574          240  
2       1621     574          237  
3       1649     585          235  
4       1670     589          226  
...      ...     ...          ...  
10093     19       7            7  
10094     18       7            7  
10095     12       3            1  
10096     13       2            1  
10097     13       2            1  

[35904 rows x 27 columns]

# A little parenthesis to analyze the national rates

In [37]:
# new row summarizing the data
df_national=df.groupby('AÑO').sum()
nat_mort=pd.read_csv("output_files/Cancer_1998_2018.csv")
#nat_mort=pd.crosstab(nat_mort.ano,nat_mort.gru_ed1,nat_mort.gru_ed1,aggfunc="count").fillna(0)
nat_mort=pd.pivot_table(nat_mort,values="gru_ed2",index="ano",columns="gru_ed1",aggfunc="count").fillna(0).reset_index().reset_index()
display(nat_mort)

df_national.columns=['ano', '0_5', '5_10',
       '10', '11', '12', '13', '14', '15', '16', '17',
       '18', '19', '20', '21', '22', '23', '24', '25',
       '26', '27', '28']
display(df_national)

"""df_mort["codmunicipio"]=df_mort.apply(lambda x: str(x.codmunre).zfill(5),axis=1)
df_mort=df_mort.astype({'ano': 'int32',"area_res":'float32'})
df_mort=pd.DataFrame(pd.pivot_table(df_mort,values='gru_ed2',index=["codmunicipio",'ano'], columns=["gru_ed1"], aggfunc="count").fillna(0).reset_index())
df_mort["mort"]=sum(df_mort[i] for i in range(10,27))

df_mort["id"]=df_mort.apply(lambda x: str(int(x.ano))+"-"+str(x.codmunicipio),axis=1)
for i in range(10,26):
    df_both["tasa_"+str(i)]=df_both[i]/df_both[f'{i}']*100000"""


gru_ed1  index     ano   10   11   12    13    14     15     16     17     18  \
0            0  1998.0  1.0  1.0  1.0  12.0  42.0   60.0  114.0  158.0  166.0   
1            1  1999.0  0.0  0.0  4.0  14.0  38.0   69.0  123.0  175.0  183.0   
2            2  2000.0  1.0  1.0  4.0  12.0  37.0   69.0  143.0  210.0  191.0   
3            3  2001.0  0.0  0.0  1.0   8.0  43.0   85.0  149.0  202.0  208.0   
4            4  2002.0  0.0  3.0  3.0  10.0  52.0   92.0  143.0  197.0  226.0   
5            5  2003.0  0.0  2.0  6.0   8.0  42.0   88.0  171.0  198.0  253.0   
6            6  2004.0  0.0  1.0  3.0  16.0  44.0   87.0  153.0  222.0  245.0   
7            7  2005.0  0.0  1.0  1.0  15.0  43.0   79.0  132.0  188.0  243.0   
8            8  2006.0  0.0  1.0  1.0  11.0  37.0  101.0  178.0  212.0  269.0   
9            9  2007.0  0.0  1.0  3.0  10.0  47.0  100.0  160.0  253.0  246.0   
10          10  2008.0  0.0  0.0  4.0  17.0  29.0   77.0  179.0  210.0  282.0   
11          11  2009.0  0.0  1.0  1.0  16.0  47.0  111.0  165.0  265.0  279.0   
12          12  2010.0  0.0  1.0  6.0  25.0  48.0  105.0  171.0  239.0  309.0   
13          13  2011.0  0.0  0.0  3.0  12.0  47.0   95.0  152.0  265.0  312.0   
14          14  2012.0  0.0  0.0  2.0  19.0  57.0  103.0  172.0  271.0  317.0   
15          15  2013.0  0.0  0.0  5.0  18.0  65.0  100.0  165.0  255.0  342.0   
16          16  2014.0  0.0  1.0  2.0  11.0  64.0   88.0  179.0  289.0  318.0   
17          17  2015.0  0.0  1.0  7.0  18.0  52.0   92.0  182.0  273.0  346.0   
18          18  2016.0  0.0  2.0  4.0  18.0  47.0  106.0  200.0  319.0  402.0   
19          19  2017.0  1.0  1.0  3.0  15.0  54.0  139.0  191.0  285.0  378.0   
20          20  2018.0  0.0  0.0  3.0  28.0  72.0  143.0  189.0  328.0  408.0   

gru_ed1     19     20     21     22     23     24     25    26  
0        147.0  131.0  121.0  103.0   99.0   67.0   59.0  45.0  
1        169.0  139.0  127.0  124.0  107.0   76.0   59.0  27.0  
2        201.0  178.0  151.0  114.0   91.0   61.0   66.0  12.0  
3        188.0  176.0  158.0  128.0  128.0   73.0   80.0   7.0  
4        210.0  187.0  138.0  158.0  109.0   89.0   86.0  13.0  
5        238.0  219.0  185.0  128.0  126.0   83.0   95.0  10.0  
6        242.0  223.0  162.0  156.0   99.0  102.0   94.0   6.0  
7        271.0  197.0  177.0  178.0  121.0  105.0  102.0   0.0  
8        235.0  198.0  175.0  166.0  130.0   95.0  101.0   0.0  
9        273.0  236.0  205.0  188.0  168.0  118.0  103.0   4.0  
10       284.0  237.0  184.0  167.0  165.0  121.0  112.0   2.0  
11       283.0  243.0  221.0  179.0  161.0  119.0  149.0   3.0  
12       324.0  283.0  231.0  198.0  172.0  125.0  152.0   5.0  
13       275.0  298.0  232.0  217.0  135.0  143.0  133.0   0.0  
14       331.0  282.0  228.0  201.0  188.0  153.0  164.0   1.0  
15       327.0  309.0  240.0  225.0  208.0  181.0  173.0   0.0  
16       383.0  329.0  276.0  242.0  194.0  180.0  217.0   0.0  
17       377.0  313.0  336.0  241.0  237.0  172.0  218.0   0.0  
18       371.0  357.0  380.0  257.0  243.0  200.0  251.0   0.0  
19       451.0  401.0  381.0  278.0  258.0  210.0  254.0   0.0  
20       425.0  399.0  351.0  312.0  237.0  230.0  303.0   0.0

ano      0_5     5_10       10       11       12       13       14  \
AÑO                                                                          
1995  42953  2122822  1987230  1907750  1753268  1621839  1621537  1531852   
1996  42953  2134433  2010799  1917984  1789738  1626162  1617178  1556757   
1997  42953  2140533  2036312  1926204  1823816  1638844  1609520  1575247   
1998  42953  2140287  2062025  1935428  1852037  1660542  1600252  1586754   
1999  42953  2135445  2083159  1949038  1872652  1690079  1591984  1591629   
2000  42953  2124243  2099716  1967556  1886116  1724343  1587450  1590593   
2001  42953  2107153  2110690  1989777  1894626  1758862  1589970  1584841   
2002  42953  2085092  2115558  2013969  1901217  1791048  1600773  1575711   
2003  42953  2059133  2113694  2038725  1909248  1817935  1621003  1565379   
2004  42953  2030109  2106111  2060062  1922289  1838010  1649874  1556673   
2005  42953  1989935  2090518  2076819  1941246  1851740  1684309  1552420   
2006  42953  1965588  2072073  2088154  1964164  1860751  1719836  1556266   
2007  42953  1942528  2050124  2093330  1989052  1867853  1752889  1568775   
2008  42953  1921443  2025968  2092494  2013803  1876269  1779964  1590221   
2009  42953  1902909  2001245  2086226  2036482  1890349  1800273  1620325   
2010  42953  1886509  1977108  2075011  2055086  1910353  1814442  1655850   
2011  42953  1870881  1954590  2058890  2067891  1933849  1823692  1691582   
2012  42953  1857348  1932584  2038750  2074115  1958526  1830958  1724071   
2013  42953  1846234  1912056  2016018  2074123  1982628  1839858  1750632   
2014  42953  1837038  1893601  1992057  2068499  2004612  1853992  1770335   
2015  42953  1830490  1878030  1969057  2058690  2024082  1874898  1785011   
2016  42953  1830415  1868502  1950413  2048100  2043719  1905225  1800412   
2017  42953  1836033  1865352  1935547  2036682  2061187  1941596  1819131   
2018  42953  1860655  1876554  1931902  2035419  2092362  1997446  1855564   
2019  42953  1897211  1902857  1940093  2039146  2129867  2067405  1910961   
2020  42953  1920187  1924862  1946045  2033561  2147776  2122489  1964503   
2021  42953  1918319  1936026  1945150  2016401  2139671  2152858  2007982   
2022  42953  1909872  1942550  1942999  1997170  2118753  2168325  2047373   
2023  42953  1898143  1948931  1942938  1981312  2094241  2175131  2085581   
2024  42953  1857339  1957470  1954589  1979831  2074954  2176639  2124929   
2025  42953  1839962  1952626  1960110  1973444  2052918  2169409  2155818   
2026  42953  1821363  1944003  1967175  1970433  2033366  2155993  2180324   

           15       16  ...       19       20       21      22      23  \
AÑO                     ...                                              
1995  1315230  1070473  ...   524161   434609   349690  256467  171388   
1996  1358925  1112775  ...   542070   443306   360033  267430  180913   
1997  1400596  1156034  ...   562502   452555   369825  279291  190385   
1998  1439533  1199870  ...   585461   463082   379040  290559  200948   
1999  1474475  1243881  ...   610728   475640   387556  300983  207675   
2000  1504213  1287272  ...   637817   490885   395575  310213  216503   
2001  1527715  1329439  ...   666383   508406   403832  318799  226025   
2002  1544789  1369616  ...   696161   528283   412912  326861  236317   
2003  1555295  1407450  ...   727058   550724   423480  334578  245715   
2004  1559803  1441874  ...   759012   575597   436172  341977  254232   
2005  1559153  1471943  ...   792186   602597   451677  349319  261658   
2006  1554740  1496452  ...   826374   631325   469266  357778  268755   
2007  1547586  1514873  ...   861639   661092   489109  367151  275447   
2008  1539287  1526548  ...   897727   691659   511176  377894  281899   
2009  1533280  1532686  ...   935274   723112   535464  390443  288245   
2010  1532295  1534483  ...   974446   755440   561549  405327  294812   
2011  1537499  1532088 

'df_mort["codmunicipio"]=df_mort.apply(lambda x: str(x.codmunre).zfill(5),axis=1)\ndf_mort=df_mort.astype({\'ano\': \'int32\',"area_res":\'float32\'})\ndf_mort=pd.DataFrame(pd.pivot_table(df_mort,values=\'gru_ed2\',index=["codmunicipio",\'ano\'], columns=["gru_ed1"], aggfunc="count").fillna(0).reset_index())\ndf_mort["mort"]=sum(df_mort[i] for i in range(10,27))\n\ndf_mort["id"]=df_mort.apply(lambda x: str(int(x.ano))+"-"+str(x.codmunicipio),axis=1)\nfor i in range(10,26):\n    df_both["tasa_"+str(i)]=df_both[i]/df_both[f\'{i}\']*100000'

In [13]:

df["yes"]=df["DPMP"].astype(int).isin(dicts.municipios_segregantes_unique.keys())
df["oldcode"]=df["DPMP"].copy()
df["DPMP"]=df["oldcode"].apply(lambda x: str(dicts.municipios_segregantes_unique[int(x)]).zfill(5) if (int(x) in dicts.municipios_segregantes_unique.keys()) else x)

df["DPMP"]=df["DPMP"].apply(lambda x: str(int(x)).zfill(5))
#df["DPMP"],df["oldcode"]=df["oldcode"].copy(),df["DPMP"].copy()


display(df[df["yes"]])
df.to_csv("población_por_grupo.csv")
df[df.DPMP!=df.oldcode]



DP    DPNOM   DPMP           MPIO   AÑO ÁREA GEOGRÁFICA  0_5  5_10  \
1720   13  Bolívar  13600         Norosí  1995           Total    0     0   
1721   13  Bolívar  13600         Norosí  1996           Total    0     0   
1722   13  Bolívar  13600         Norosí  1997           Total    0     0   
1723   13  Bolívar  13600         Norosí  1998           Total    0     0   
1724   13  Bolívar  13600         Norosí  1999           Total    0     0   
...    ..      ...    ...            ...   ...             ...  ...   ...   
10057  97   Vaupés  97001  Yavaraté (CD)  2022           Total   57   111   
10058  97   Vaupés  97001  Yavaraté (CD)  2023           Total   61   113   
10059  97   Vaupés  97001  Yavaraté (CD)  2024           Total   60   113   
10060  97   Vaupés  97001  Yavaraté (CD)  2025           Total   60   116   
10061  97   Vaupés  97001  Yavaraté (CD)  2026           Total   58   119   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
1720       0      0  ...      0      0      0      0      0      0       0   
1721       0      0  ...      0      0      0      0      0      0       0   
1722       0      0  ...      0      0      0      0      0      0       0   
1723       0      0  ...      0      0      0      0      0      0       0   
1724       0      0  ...      0      0      0      0      0      0       0   
...      ...    ...  ...    ...    ...    ...    ...    ...    ...     ...   
10057    129    104  ...      4      4      7     12      3      4       2   
10058    131    111  ...      5      4      7     12      3      4       3   
10059    141    114  ...     13     11      8      5      5      1       0   
10060    150    124  ...     15     11      9      5      5      1       0   
10061    158    132  ...     15     12      9      5      5      0       0   

       Mujeres_100   yes  oldcode  
1720             0  True    13490  
1721             0  True    13490  
1722             0  True    13490  
1723             0  True    13490  
1724             0  True    13490  
...            ...   ...      ...  
10057            0  True    97889  
10058            0  True    97889  
10059            0  True    97889  
10060            0  True    97889  
10061            0  True    97889  

[1920 rows x 29 columns]

DP    DPNOM   DPMP           MPIO   AÑO ÁREA GEOGRÁFICA  0_5  5_10  \
1720   13  Bolívar  13600         Norosí  1995           Total    0     0   
1721   13  Bolívar  13600         Norosí  1996           Total    0     0   
1722   13  Bolívar  13600         Norosí  1997           Total    0     0   
1723   13  Bolívar  13600         Norosí  1998           Total    0     0   
1724   13  Bolívar  13600         Norosí  1999           Total    0     0   
...    ..      ...    ...            ...   ...             ...  ...   ...   
10057  97   Vaupés  97001  Yavaraté (CD)  2022           Total   57   111   
10058  97   Vaupés  97001  Yavaraté (CD)  2023           Total   61   113   
10059  97   Vaupés  97001  Yavaraté (CD)  2024           Total   60   113   
10060  97   Vaupés  97001  Yavaraté (CD)  2025           Total   60   116   
10061  97   Vaupés  97001  Yavaraté (CD)  2026           Total   58   119   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
1720       0      0  ...      0      0      0      0      0      0       0   
1721       0      0  ...      0      0      0      0      0      0       0   
1722       0      0  ...      0      0      0      0      0      0       0   
1723       0      0  ...      0      0      0      0      0      0       0   
1724       0      0  ...      0      0      0      0      0      0       0   
...      ...    ...  ...    ...    ...    ...    ...    ...    ...     ...   
10057    129    104  ...      4      4      7     12      3      4       2   
10058    131    111  ...      5      4      7     12      3      4       3   
10059    141    114  ...     13     11      8      5      5      1       0   
10060    150    124  ...     15     11      9      5      5      1       0   
10061    158    132  ...     15     12      9      5      5      0       0   

       Mujeres_100   yes  oldcode  
1720             0  True    13490  
1721             0  True    13490  
1722             0  True    13490  
1723             0  True    13490  
1724             0  True    13490  
...            ...   ...      ...  
10057            0  True    97889  
10058            0  True    97889  
10059            0  True    97889  
10060            0  True    97889  
10061            0  True    97889  

[1920 rows x 29 columns]

In [14]:
df[(df.DPMP=="97001") & (df["AÑO"]==1998)]

DP   DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA   0_5  5_10  \
11123  97  Vaupés  97001            Mitú  1998           Total  1083   873   
11143  97  Vaupés  97001      Pacoa (CD)  1998           Total   149   119   
11163  97  Vaupés  97001  Papunahua (CD)  1998           Total    33    27   
11173  97  Vaupés  97001   Yavaraté (CD)  1998           Total    42    32   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
11123    744    551  ...     81     59     47     36     23     20      14   
11143    111     87  ...      6      6      5      3      3      1       0   
11163     21     15  ...      5      5      5      5      3      0       0   
11173     30     22  ...      6      5      6      5      5      2       0   

       Mujeres_100    yes  oldcode  
11123            6  False    97001  
11143            0   True    97511  
11163            0   True    97777  
11173            0   True    97889  

[4 rows x 29 columns]

# Sumamos las poblaciones de los municipios que componen cada una de las nuevas regiones conformadas.

In [15]:
df=df.query("AÑO>=1998 and AÑO<=2018").copy()

In [16]:
for ids in df[df["yes"]].DPMP.unique():
    for year in range(1998,2019):
            for i in range(20):
                df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), str(5*i)+"_"+str(5*(i+1))]=df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), str(5*i)+"_"+str(5*(i+1))]+sum(df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (df["yes"]), str(5*i)+"_"+str(5*(i+1))])
            df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), "Mujeres_100"]=df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), "Mujeres_100"]+sum(df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (df["yes"]),  "Mujeres_100"])




In [17]:
df[(df.DPMP=="97001") & (df["AÑO"]==1998)]

DP   DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA   0_5  5_10  \
11123  97  Vaupés  97001            Mitú  1998           Total  1307  1051   
11143  97  Vaupés  97001      Pacoa (CD)  1998           Total   149   119   
11163  97  Vaupés  97001  Papunahua (CD)  1998           Total    33    27   
11173  97  Vaupés  97001   Yavaraté (CD)  1998           Total    42    32   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
11123    906    675  ...     98     75     63     49     34     23      14   
11143    111     87  ...      6      6      5      3      3      1       0   
11163     21     15  ...      5      5      5      5      3      0       0   
11173     30     22  ...      6      5      6      5      5      2       0   

       Mujeres_100    yes  oldcode  
11123            6  False    97001  
11143            0   True    97511  
11163            0   True    97777  
11173            0   True    97889  

[4 rows x 29 columns]

# Después de sumar, conservamos solo los municipios originales

In [18]:
df=df[~df['yes']].copy()

In [19]:
del df['yes']
#del df["newcode"]

In [20]:
df["Total"]=sum(df[str(5*i)+"_"+str(5*(i+1))] for i in range(20))+df["Mujeres_100"]


In [21]:
df_try=pd.DataFrame(pd.pivot_table(df[(df["AÑO"]>=1998) & (df["AÑO"]<=2018)],values='Total',index=["DPMP"], columns=["AÑO"], aggfunc="sum").fillna(0).reset_index())
df_try.to_csv("pob_recodificada.csv")


In [22]:
df_try[df_try.DPMP=="97001"]

AÑO    DPMP  1998  1999  2000  2001  2002  2003  2004   2005   2006  ...  \
1055  97001  7357  7700  8044  8440  8839  9293  9755  10223  10745  ...   

AÑO    2009   2010   2011   2012   2013   2014   2015   2016   2017   2018  
1055  12375  12931  13450  13985  14474  15000  15503  15999  16471  16966  

[1 rows x 22 columns]

## Limiting the dataframe to contain information only for the time period being considered in this research

In [23]:
df=df[df["AÑO"].isin([ano for ano in range(1998,2019)])].copy()
df=df[~df.DPMP.isin(["880001","88564"])].copy()

## Changing the name of the columns to correspond to those of the Cancer datasets (age group)

In [24]:
df.columns

Index(['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', '0_5', '5_10',
       '10_15', '15_20', '20_25', '25_30', '30_35', '35_40', '40_45', '45_50',
       '50_55', '55_60', '60_65', '65_70', '70_75', '75_80', '80_85', '85_90',
       '90_95', '95_100', 'Mujeres_100', 'oldcode', 'Total'],
      dtype='object')

In [25]:
df.columns=['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', '0_5', '5_10',
       '10', '11', '12', '13', '14', '15', '16', '17',
       '18', '19', '20', '21', '22', '23', '24', '25',
       '26', '27', '28','oldcode','Total']

# All information for age groups over 25 are associated (summed up) to the latter.

In [26]:
df["25"]=sum([df["25"],df["26"],df["27"],df["28"]])

# Keeping information for the age groups of interest

In [27]:
df=df[['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', 
       '10', '11', '12', '13', '14', '15', '16', '17',
       '18', '19', '20', '21', '22', '23', '24', '25','Total']].copy()
df

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
3       5  Antioquia  05001        Medellín  1998           Total  86309   
4       5  Antioquia  05001        Medellín  1999           Total  87795   
5       5  Antioquia  05001        Medellín  2000           Total  89643   
6       5  Antioquia  05001        Medellín  2001           Total  91471   
7       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
10035  97     Vaupés  97666         Taraira  2018           Total    161   
10062  99    Vichada  99001  Puerto Carreño  2018           Total    977   
10071  99    Vichada  99524    La Primavera  2018           Total    585   
10080  99    Vichada  99624   Santa Rosalía  2018           Total    253   
10089  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...     17     18     19     20     21     22  \
3      87523  82886  85891  ...  56287  45277  34023  26166  21949  17021   
4      89112  84462  84151  ...  58570  47328  35824  26943  22456  17727   
5      90256  86793  82865  ...  60913  49388  37728  27968  22744  18119   
6      90985  89390  82136  ...  63397  51486  39748  29282  22993  18381   
7      91592  91773  82112  ...  65961  53683  41834  30743  23518  18858   
...      ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   
10035    140     78     62  ...     38     25     16     15      8      5   
10062    950    922    889  ...    466    402    337    275    178     79   
10071    483    401    392  ...    234    198    142    108     87     49   
10080    200    162    142  ...     88     77     64     53     39     20   
10089   4016   3063   2416  ...   1215   1004    795    528    385    233   

          23    24    25    Total  
3      12315  7430  6393   991843  
4      12577  7974  6584  1008226  
5      12915  8382  6784  1024009  
6      13284  8820  7044  1039108  
7      13799  9254  7355  1053805  
...      ...   ...   ...      ...  
10035      6     3     7     1066  
10062     47    26    30     9912  
10071     27    17    27     4724  
10080     13     8     7     1929  
10089    139    81    71    34001  

[22281 rows x 23 columns]

# Creating an auxiliar ID that will be later used for joining the population data set to the Cancer data set

In [28]:
df["id"]=df.apply(lambda x: str(x.AÑO)+"-"+str(x.DPMP),axis=1)

In [35]:
df[(df.AÑO==1998) & (df.DPMP=="94663")]

DP    DPNOM   DPMP             MPIO   AÑO ÁREA GEOGRÁFICA   10   11  \
11013  94  Guainía  94663  Mapiripana (CD)  1998           Total  286  216   

        12   13  ...  18  19  20  21  22  23  24  25  Total          id  
11013  129  100  ...  23  13   2  10  10   7   0   0   1815  1998-94663  

[1 rows x 24 columns]

# Reading the Cancer data set and Joining Tables

In [36]:
df_mort=pd.read_csv("output_files/Cancer_1998_2018.csv")


In [37]:
df_mort["codmunicipio"]=df_mort.apply(lambda x: str(x.codmunre).zfill(5),axis=1)
df_mort=df_mort.astype({'ano': 'int32',"area_res":'float32'})
df_mort=pd.DataFrame(pd.pivot_table(df_mort,values='gru_ed2',index=["codmunicipio",'ano'], columns=["gru_ed1"], aggfunc="count").fillna(0).reset_index())



In [38]:
# se Deja hasta el 27, porque si bien no se sabe la edad, si se sabe el año y municipio de la muerte

df_mort["mort"]=sum(df_mort[i] for i in range(10,27))


In [39]:
del df_mort[26]
df_mort.mort.sum()

46672.0

# Se observan 132 casos de fallecidos para los cuales no se conocer la edad, incorrectamente los estabamos sumando antes al grupo de mayor edad.

df_mort=df_mort[~df_mort.codmunicipio.isin(excluded)].copy()

In [40]:
df_mort.columns

Index(['codmunicipio',          'ano',             10,             11,
                   12,             13,             14,             15,
                   16,             17,             18,             19,
                   20,             21,             22,             23,
                   24,             25,         'mort'],
      dtype='object', name='gru_ed1')

# Creating an auxiliar ID to join DataFrames

In [41]:
df_mort["id"]=df_mort.apply(lambda x: str(int(x.ano))+"-"+str(x.codmunicipio),axis=1)


In [42]:
df_both=df.merge(df_mort,how="left",left_on=["DPMP",'AÑO'],right_on=["codmunicipio","ano"])
df_both.to_excel("output_files/revisame.xlsx")


In [43]:
df_both.mort.sum()

46672.0

## Computing the mortality rates for each group

In [44]:
for i in range(10,26):
    df_both["tasa_"+str(i)]=df_both[i]/df_both[f'{i}']*100000


In [45]:
df_both.sort_values(by=["DPMP","AÑO"]).head(50)

DP      DPNOM   DPMP       MPIO   AÑO ÁREA GEOGRÁFICA     10      11  \
0       5  Antioquia  05001   Medellín  1998           Total  86309   87523   
1       5  Antioquia  05001   Medellín  1999           Total  87795   89112   
2       5  Antioquia  05001   Medellín  2000           Total  89643   90256   
3       5  Antioquia  05001   Medellín  2001           Total  91471   90985   
4       5  Antioquia  05001   Medellín  2002           Total  93120   91592   
5       5  Antioquia  05001   Medellín  2003           Total  94451   92407   
6       5  Antioquia  05001   Medellín  2004           Total  95450   93538   
7427    5  Antioquia  05001   Medellín  2005           Total  92223   93032   
7428    5  Antioquia  05001   Medellín  2006           Total  92454   94632   
7429    5  Antioquia  05001   Medellín  2007           Total  92327   96208   
7430    5  Antioquia  05001   Medellín  2008           Total  91873   97578   
7431    5  Antioquia  05001   Medellín  2009           Total  91145   98740   
7432    5  Antioquia  05001   Medellín  2010           Total  90148   99627   
7433    5  Antioquia  05001   Medellín  2011           Total  88995  100286   
7434    5  Antioquia  05001   Medellín  2012           Total  87777  100688   
7435    5  Antioquia  05001   Medellín  2013           Total  86407  100645   
7436    5  Antioquia  05001   Medellín  2014           Total  84901  100095   
7437    5  Antioquia  05001   Medellín  2015           Total  83421   99071   
7438    5  Antioquia  05001   Medellín  2016           Total  82185   97859   
7439    5  Antioquia  05001   Medellín  2017           Total  81145   96580   
21220   5  Antioquia  05001   Medellín  2018           Total  80380   95337   
7       5  Antioquia  05002  Abejorral  1998           Total   1292    1159   
8       5  Antioquia  05002  Abejorral  1999           Total   1246    1106   
9       5  Antioquia  05002  Abejorral  2000           Total   1209    1044   
10      5  Antioquia  05002  Abejorral  2001           Total   1191    1009   
11      5  Antioquia  05002  Abejorral  2002           Total   1180     988   
12      5  Antioquia  05002  Abejorral  2003           Total   1163     970   
13      5  Antioquia  05002  Abejorral  2004           Total   1153     959   
7440    5  Antioquia  05002  Abejorral  2005           Total   1147     952   
7441    5  Antioquia  05002  Abejorral  2006           Total   1125     949   
7442    5  Antioquia  05002  Abejorral  2007           Total   1102     946   
7443    5  Antioquia  05002  Abejorral  2008           Total   1071     939   
7444    5  Antioquia  05002  Abejorral  2009           Total   1044     932   
7445    5  Antioquia  05002  Abejorral  2010           Total   1016     927   
7446    5  Antioquia  05002  Abejorral  2011           Total    988     922   
7447    5  Antioquia  05002  Abejorral  2012           Total    960     912   
7448    5  Antioquia  05002  Abejorral  2013           Total    931     900   
7449    5  Antioquia  05002  Abejorral  2014           Total    892     872   
7450    5  Antioquia  05002  Abejorral  2015           Total    856     848   
7451    5  Antioquia  05002  Abejorral  2016           Total    831     820   
7452    5  Antioquia  05002  Abejorral  2017           Total    803     793   
21221   5  Antioquia  05002  Abejorral  2018           Total    782     781   
14      5  Antioquia  05004   Abriaquí  1998           Total    136     129   
15      5  Antioquia  05004   Abriaquí  1999           Total    140     122   
16      5  Antioquia  05004   Abriaquí  2000           Total    140     120   
17      5  Antioquia  05004   Abriaquí  2001           Total    140     113   
18      5  Antioquia  05004   Abriaquí  2002           Total    142     115   
19      5  Antioquia  05004   Abriaquí  2003           Total    139     107   
20      5  Antioquia  05004   Abriaquí  2004           Total    136     108   
7453    5  Antioquia  05004   Abriaquí  2005           To

In [46]:
df_both=df_both.fillna(0)
df_both.to_csv("output_files/TasasxGrupoEdad.csv")

## Creating a dictionary based on the weights corresponding to each age group according to the WHO (2000 - 2025) Note: Does not sum 100 because it was not necessary to keep the age groups below 10, provided there were no information about breast cancer in the studied period

df_both=pd.read_csv("output_files/TasasxGrupoEdad.csv")

In [47]:
diccionario={10: 8.60,
11: 8.47,
12: 8.22,
13: 7.93,
14: 7.61,
15: 7.15,
16: 6.59,
17 :6.04,
18 :5.37,
19 :4.55,
20 :3.72,
21 :2.96,
22 :2.21,
23 :1.52,
24 :0.91,
25 :0.44+0.15+0.04+0.005}

In [48]:
df_both=df_both.fillna(0)


### Adjusting mortality rate by age (direct method)

In [49]:
df_both["TAE"]=sum(df_both["tasa_"+str(i)]*diccionario[i]/100 for i in range(10,26))
df_both["TAE"]

0        14.004688
1        16.097837
2        16.623773
3        12.775930
4        16.271970
           ...    
22276     0.000000
22277     0.000000
22278     0.000000
22279     0.000000
22280     0.000000
Name: TAE, Length: 22281, dtype: float64

In [50]:
df_both

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
0       5  Antioquia  05001        Medellín  1998           Total  86309   
1       5  Antioquia  05001        Medellín  1999           Total  87795   
2       5  Antioquia  05001        Medellín  2000           Total  89643   
3       5  Antioquia  05001        Medellín  2001           Total  91471   
4       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
22276  97     Vaupés  97666         Taraira  2018           Total    161   
22277  99    Vichada  99001  Puerto Carreño  2018           Total    977   
22278  99    Vichada  99524    La Primavera  2018           Total    585   
22279  99    Vichada  99624   Santa Rosalía  2018           Total    253   
22280  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...    tasa_17    tasa_18    tasa_19    tasa_20  \
0      87523  82886  85891  ...  21.319310  37.546657  23.513506  64.969808   
1      89112  84462  84151  ...  35.854533  29.580798  64.202769  29.692313   
2      90256  86793  82865  ...  32.833714  40.495667  58.312129  57.208238   
3      90985  89390  82136  ...  25.237787  23.307307  47.801147  23.905471   
4      91592  91773  82112  ...  22.740710  27.941807  40.636803  68.308233   
...      ...    ...    ...  ...        ...        ...        ...        ...   
22276    140     78     62  ...   0.000000   0.000000   0.000000   0.000000   
22277    950    922    889  ...   0.000000   0.000000   0.000000   0.000000   
22278    483    401    392  ...   0.000000   0.000000   0.000000   0.000000   
22279    200    162    142  ...   0.000000   0.000000   0.000000   0.000000   
22280   4016   3063   2416  ...   0.000000   0.000000   0.000000   0.000000   

         tasa_21    tasa_22     tasa_23     tasa_24     tasa_25        TAE  
0      63.784227  64.626050  105.562322   80.753701   62.568434  14.004688  
1      62.344140  78.975574   95.412260  112.866817   91.130012  16.097837  
2      35.174112  55.190684   61.943477  131.233596  147.405660  16.623773  
3      52.189797  43.523203  105.389943   56.689342   85.178876  12.775930  
4      46.772685  95.450207   94.209725  129.673655  163.154317  16.271970  
...          ...        ...         ...         ...         ...        ...  
22276   0.000000   0.000000    0.000000    0.000000    0.000000   0.000000  
22277   0.000000   0.000000    0.000000    0.000000    0.000000   0.000000  
22278   0.000000   0.000000    0.000000    0.000000    0.000000   0.000000  
22279   0.000000   0.000000    0.000000    0.000000    0.000000   0.000000  
22280   0.000000   0.000000    0.000000    0.000000    0.000000   0.000000  

[22281 rows x 61 columns]

df_out=df_both[df_both["AÑO"]==2018][["DPMP","TEA"]].copy()
df_out.DPMP=df_out.DPMP.apply(lambda x: str(x).zfill(5) )
df_out.to_csv("TEA_2018.csv")
df_out


# Computando la tasa ajustada de mortalidad por quinquenio

Agregando una variable llamada periodo de acuerdo al año evaluado.

In [51]:
df_both["periodo"]=pd.cut(df_both.AÑO, bins=[0, 2002, 2007, 2012, 100000], labels='1998-2002 2003-2007 2008-2012 2013-2018'.split())
df_both=df_both[['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA','periodo','TAE',"mort"]].copy()
df_both.DPMP=df_both.DPMP.apply(lambda x: str(int(x)).zfill(5))
df_both

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA    periodo  \
0       5  Antioquia  05001        Medellín  1998           Total  1998-2002   
1       5  Antioquia  05001        Medellín  1999           Total  1998-2002   
2       5  Antioquia  05001        Medellín  2000           Total  1998-2002   
3       5  Antioquia  05001        Medellín  2001           Total  1998-2002   
4       5  Antioquia  05001        Medellín  2002           Total  1998-2002   
...    ..        ...    ...             ...   ...             ...        ...   
22276  97     Vaupés  97666         Taraira  2018           Total  2013-2018   
22277  99    Vichada  99001  Puerto Carreño  2018           Total  2013-2018   
22278  99    Vichada  99524    La Primavera  2018           Total  2013-2018   
22279  99    Vichada  99624   Santa Rosalía  2018           Total  2013-2018   
22280  99    Vichada  99773        Cumaribo  2018           Total  2013-2018   

             TAE   mort  
0      14.004688  117.0  
1      16.097837  140.0  
2      16.623773  152.0  
3      12.775930  121.0  
4      16.271970  156.0  
...          ...    ...  
22276   0.000000    0.0  
22277   0.000000    0.0  
22278   0.000000    0.0  
22279   0.000000    0.0  
22280   0.000000    0.0  

[22281 rows x 9 columns]

df_out# Se genera un nuevo dataframe que asocia municipios, años y tasas de mortalidad para posteriormente unirlo con las variables socioeconómicas.

In [52]:
df_both.mort.sum()

46672.0

In [53]:
df

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
3       5  Antioquia  05001        Medellín  1998           Total  86309   
4       5  Antioquia  05001        Medellín  1999           Total  87795   
5       5  Antioquia  05001        Medellín  2000           Total  89643   
6       5  Antioquia  05001        Medellín  2001           Total  91471   
7       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
10035  97     Vaupés  97666         Taraira  2018           Total    161   
10062  99    Vichada  99001  Puerto Carreño  2018           Total    977   
10071  99    Vichada  99524    La Primavera  2018           Total    585   
10080  99    Vichada  99624   Santa Rosalía  2018           Total    253   
10089  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...     18     19     20     21     22     23  \
3      87523  82886  85891  ...  45277  34023  26166  21949  17021  12315   
4      89112  84462  84151  ...  47328  35824  26943  22456  17727  12577   
5      90256  86793  82865  ...  49388  37728  27968  22744  18119  12915   
6      90985  89390  82136  ...  51486  39748  29282  22993  18381  13284   
7      91592  91773  82112  ...  53683  41834  30743  23518  18858  13799   
...      ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   
10035    140     78     62  ...     25     16     15      8      5      6   
10062    950    922    889  ...    402    337    275    178     79     47   
10071    483    401    392  ...    198    142    108     87     49     27   
10080    200    162    142  ...     77     64     53     39     20     13   
10089   4016   3063   2416  ...   1004    795    528    385    233    139   

         24    25    Total          id  
3      7430  6393   991843  1998-05001  
4      7974  6584  1008226  1999-05001  
5      8382  6784  1024009  2000-05001  
6      8820  7044  1039108  2001-05001  
7      9254  7355  1053805  2002-05001  
...     ...   ...      ...         ...  
10035     3     7     1066  2018-97666  
10062    26    30     9912  2018-99001  
10071    17    27     4724  2018-99524  
10080     8     7     1929  2018-99624  
10089    81    71    34001  2018-99773  

[22281 rows x 24 columns]

In [54]:
df_mort.ano.unique()

array([1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008,
       2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018])

In [55]:
df_both.groupby(["DPMP","AÑO"]).agg("count")[["TAE"]].reset_index().query("TAE==1")

DPMP   AÑO  TAE
0      05001  1998    1
1      05001  1999    1
2      05001  2000    1
3      05001  2001    1
4      05001  2002    1
...      ...   ...  ...
22276  99773  2014    1
22277  99773  2015    1
22278  99773  2016    1
22279  99773  2017    1
22280  99773  2018    1

[22281 rows x 3 columns]

In [56]:
df_out=pd.DataFrame(df.DPMP.astype(int).unique())
df_out.columns=["DPMP"]
df_out.DPMP=df_out.DPMP.apply(lambda x: str(int(x)).zfill(5))



In [57]:
df_both[df_both["AÑO"]==2005].set_index("DPMP")["mort"]

DPMP
05001    138.0
05002      2.0
05004      0.0
05021      0.0
05030      1.0
         ...  
97666      0.0
99001      1.0
99524      0.0
99624      0.0
99773      0.0
Name: mort, Length: 1061, dtype: float64

In [58]:
for y in list(df_both[df_both["AÑO"]>0]["AÑO"].unique()):
    df_out["mort_"+str(y)]=df_out.DPMP.map(df_both[df_both["AÑO"]==y].set_index("DPMP")["mort"])    
    
for y in list(df_both[df_both["AÑO"]>0]["AÑO"].unique()):
    df_out["TAE_"+str(y)]=df_out.DPMP.map(df_both[df_both["AÑO"]==y].set_index("DPMP")["TAE"])



df_out.DPMP=df_out.DPMP.apply(lambda x: str(int(x)).zfill(5))
df_out=df_out.fillna(0)

In [59]:
dictio=pd.DataFrame(pd.read_csv("data/DIVIPOLA_Municipios.csv"))
dictio.COD=dictio.COD.astype(str)
dictio.COD=dictio.COD.apply(lambda x: x.zfill(5))
print(dictio.columns)
dictio=dictio.set_index('COD').T.to_dict()
for i in dictio.keys():
    dictio[i]=dictio[i]['MUNICIPIO']

df_out["MPIO"]=df_out.DPMP.map(dictio).apply(lambda x: str(x).capitalize())

Index(['COD', 'MUNICIPIO'], dtype='object')


In [60]:
df_out.to_csv("output_files/TAE_&_mortality.csv")

In [61]:
df_out.to_excel("output_files/TAE_&_mortality.xlsx")

In [62]:
df_out

DPMP  mort_1998  mort_1999  mort_2000  mort_2001  mort_2002  mort_2003  \
0     05001      117.0      140.0      152.0      121.0      156.0      181.0   
1     05002        0.0        0.0        1.0        0.0        0.0        0.0   
2     05004        0.0        0.0        0.0        0.0        0.0        0.0   
3     05021        1.0        0.0        0.0        0.0        0.0        0.0   
4     05030        0.0        1.0        2.0        0.0        0.0        1.0   
...     ...        ...        ...        ...        ...        ...        ...   
1056  97666        0.0        0.0        0.0        0.0        0.0        0.0   
1057  99001        0.0        2.0        1.0        0.0        0.0        0.0   
1058  99524        0.0        0.0        0.0        0.0        0.0        0.0   
1059  99624        0.0        0.0        0.0        0.0        0.0        0.0   
1060  99773        0.0        0.0        0.0        0.0        0.0        0.0   

      mort_2004  mort_2005  mort_2006  ...   TAE_2010   TAE_2011   TAE_2012  \
0         166.0      138.0      181.0  ...  15.113675  15.304540  15.472662   
1           1.0        2.0        0.0  ...   8.551724   0.000000   0.000000   
2           0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
3           0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
4           0.0        1.0        1.0  ...  19.424761  24.301651   5.994730   
...         ...        ...        ...  ...        ...        ...        ...   
1056        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
1057        1.0        1.0        0.0  ...  41.081081   0.000000   0.000000   
1058        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
1059        0.0        0.0        0.0  ...   0.000000  70.860215   0.000000   
1060        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   

       TAE_2013    TAE_2014   TAE_2015   TAE_2016   TAE_2017   TAE_2018  \
0     15.775365   15.975835  14.570198  16.403328  16.935918  14.176541   
1     18.048366    7.322835   9.263804  28.786943   7.106109   0.000000   
2      0.000000  139.085297  69.062500   0.000000  82.222222   0.000000   
3      0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
4      6.757991   13.481400   5.118110  18.817483  17.079392   4.846570   
...         ...         ...        ...        ...        ...        ...   
1056   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1057   0.000000    0.000000   0.000000  14.583333  16.723164   0.000000   
1058   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1059   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1060   0.000000    0.000000   0.000000   5.354610   6.034483   0.000000   

                MPIO  
0           Medellín  
1          Abejorral  
2           Abriaquí  
3         Alejandría  
4              Amagá  
...              ...  
1056         Taraira  
1057  Puerto carreño  
1058    La primavera  
1059   Santa rosalía  
1060        Cumaribo  

[1061 rows x 44 columns]